packages/cli/src/commands/build.ts

```diff
@@ -0,0 +1,76 @@
+import { Command } from 'commander';
+import consola from 'consola';
+import { cp, mkdtemp, rm } from 'node:fs/promises';
+import { tmpdir } from 'node:os';
+import { join, relative, sep } from 'node:path';
+
+const SKIP_DIRS = new Set([
+  'node_modules',
+  '.bigcommerce',
+  '.git',
+  '.turbo',
+  '.next',
+  '.vscode',
+  '.github',
+  '.changeset',
+  'dist',
+]);
+
+export function createFilter(root: string, skipDirs: Set<string>) {
+  return (src: string) => {
+    const rel = relative(root, src);
+    const parts = rel.split(sep);
+
+    return !parts.some((part) => skipDirs.has(part));
+  };
+}
+
+export async function mkTempDir() {
+  const prefix = join(tmpdir(), 'catalyst-build-');
+  const path = await mkdtemp(prefix);
+
+  consola.info(`Created temporary directory: ${path}`);
+
+  return [
+    path,
+    async () => {
+      try {
+        consola.info(`Cleaning up temporary directory: ${path}`);
+        await rm(path, { recursive: true, force: true });
+        consola.success('Cleanup complete');
+      } catch (error) {
+        consola.warn(`Failed to clean up temporary directory: ${path}`, error);
+      }
+    },
+  ] as const;
+}
+
+export const build = new Command('build')
+  .option('--keep-temp-dir', 'Keep the temporary directory after the build')
```

Comment:
Should we set the pattern to use `kebab case` for all options? In preference over `camel case`?

Reply:
Yeah, POSIX [standardized](https://www.gnu.org/prep/standards/html_node/Option-Table.html) on `--kebab-case` convention, so it's what people are used to IMO. Commander [automatically transforms those to camel case too](https://github.com/tj/commander.js?tab=readme-ov-file#options):
> Multi-word options such as "--template-engine" are camel-cased, becoming `program.opts().templateEngine` etc.